docs/en/operations/storing-data.md

```diff
@@ -172,37 +172,41 @@ Example of configuration for versions earlier than 22.8:
+- `enable_cache_hits_threshold` - number which defines how many times some data needs to be read before it will be cached. Default: `0`, e.g. the data is cached at the first attempt to read it.
 
 - `do_not_evict_index_and_mark_files` - do not evict small frequently used files according to cache policy. Default: `false`. This setting was added in version 22.8. If you used filesystem cache before this version, then it will not work on versions starting from 22.8 if this setting is set to `true`. If you want to use this setting, clear old cache created before version 22.8 before upgrading.
 
 - `max_file_segment_size` - a maximum size of a single cache file in bytes or in readable format (`ki, Mi, Gi, etc`, example `10Gi`). Default: `104857600` (`100Mi`).
 
 - `max_elements` - a limit for a number of cache files. Default: `1048576`.
 
-Cache **query settings**:
+File Cache **query settings**:
+
+Some of these settings will disable cache per query features that are enabled by default. For example, setting `cache_on_write_operations` to 1 means that general file and per query cache are enabled but also setting `enable_filesystem_cache_on_write_operations` to 1 means that file cache is enabled but disabled per query cache. The same approach must be used for `enable_filesystem_cache`, if set to 1, cache per query is disabled.
 
-- `enable_filesystem_cache` - allows to disable cache per query even if storage policy was configured with `cache` disk type. Default: `true`.
+- `enable_filesystem_cache` - allows to disable cache per query even if storage policy was configured with `cache` disk type. Default: `true`. This setting should be defined in the disk configuration section.
 
-- `read_from_filesystem_cache_if_exists_otherwise_bypass_cache` - allows to use cache in query only if it already exists, otherwise query data will not be written to local cache storage. Default: `false`.
+- `read_from_filesystem_cache_if_exists_otherwise_bypass_cache` - allows to use cache in query only if it already exists, otherwise query data will not be written to local cache storage. Default: `false`. This is a profile level configuration setting.
 
-- `enable_filesystem_cache_on_write_operations` - turn on `write-through` cache. This setting works only if setting `cache_on_write_operations` in cache configuration is turned on.
+- `enable_filesystem_cache_on_write_operations` - turn on `write-through` cache. This setting works only if setting `cache_on_write_operations` in cache configuration is turned on. This setting should be defined in the disk configuration section.
 
-- `enable_filesystem_cache_log` - turn on logging to `system.filesystem_cache_log` table. Gives a detailed view of cache usage per query. Default: `false`.
+- `enable_filesystem_cache_log` - turn on logging to `system.filesystem_cache_log` table. Gives a detailed view of cache usage per query. Default: `false`. This is a profile level configuration setting.
```

Comment:
Now I understand thanks for clarifying, the query settings are referring to per query settings and can be also used in a profile. Because I understood (and other people also) that query settings were to disable per query cache but were part of the disk configuration.

So I'l rewrite the things that you pointed out and also, specify:

File Cache **disk configuration settings**:
File Cache **query/profile settings**:

Also checking system.settings some descriptions also were a little bit difficult to understand comparing :

```
`enable_filesystem_cache` - allows to disable cache per query even if storage policy was configured with `cache` disk type. Default: `true`
```

from documentation with descriptions from system.settings, and led to some misunderstandings.

```
SELECT
    name,
    value,
    description
FROM system.settings
WHERE name ILIKE '%filesystem_cache%'

Query id: 924432cf-9fb1-48f2-b7d7-f01a20a35f88

┌─name────────────────────────────────────────────────────────┬─value────────┬─description────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ enable_filesystem_cache                                     │ 1            │ Use cache for remote filesystem. This setting does not turn on/off cache for disks (must be done via disk config), but allows to bypass cache for some queries if intended                                                                                 │
│ enable_filesystem_cache_on_write_operations                 │ 0            │ Write into cache on write operations. To actually work this setting requires be added to disk config too                                                                                                                                                   │
│ enable_filesystem_cache_log                                 │ 0            │ Allows to record the filesystem caching log for each query                                                                                                                                                                                                 │
│ read_from_filesystem_cache_if_exists_otherwise_bypass_cache │ 0            │ Allow to use the filesystem cache in passive mode - benefit from the existing cache entries, but don't put more entries into the cache. If you set this setting for heavy ad-hoc queries and leave it disabled for short real-time queries, this will allows to avoid cache threshing by too heavy queries and to improve the overall system efficiency. │
│ enable_filesystem_cache_on_lower_level                      │ 1            │ If read buffer supports caching inside threadpool, allow it to do it, otherwise cache outside ot threadpool. Do not use this setting, it is needed for testing                                                                                             │
│ filesystem_cache_max_download_size                          │ 137438953472 │ Max remote filesystem cache size that can be downloaded by a single query                                                                                                                                                                                  │
└─────────────────────────────────────────────────────────────┴──────────────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```